### PR TITLE
fix(Carousel): SSR Layout shift issues

### PR DESCRIPTION
**Pull request by atabel**
WEB-1672

### CODE REVIEW

File: src/carousel.tsx [hunk 1]
```diff
@@ -45,20 +43,25 @@ export const PageBullets: React.FC<PageBulletsProps> = ({currentIndex, numPages,
...
-                        border: 'none',
-                        background: 'transparent',
-                    })}
+                    className={classNames(
+                        typeof numPages === 'number'
+                            ? styles.bulletButton
+                            : {
+                                  [styles.bulletButtonMobile]: i < numPages.mobile,
+                                  [styles.bulletButtonTablet]: i < (numPages.tablet ?? numPages.mobile),
+                                  [styles.bulletButtonDesktop]: i < numPages.desktop,
```
Comment (atabel) [inline src/carousel.tsx:61]: the number of pages/bullets depends on breakpoints (mobile/tablet/desktop)

File: src/carousel.tsx [hunk 2]
```diff
@@ -395,7 +409,8 @@ const BaseCarousel: React.FC<BaseCarouselProps> = ({
                 <div
                     className={classNames(styles.carousel, {
                         [styles.centeredCarousel]: centered,
-                        [styles.carouselWithScroll]: pagesCount > 1,
+                        [styles.carouselWithScrollMobile]: pagesCountMobile > 1,
+                        [styles.carouselWithScrollTablet]: pagesCountTablet > 1,
```
Comment (atabel) [inline src/carousel.tsx:413]: the carousel may have scroll in one breakpoint

File: src/test-utils/index.tsx [hunk 3]
```diff
@@ -103,8 +103,8 @@ const DEVICES: DeviceCollection = {
             width: 800,
             height: 600,
             deviceScaleFactor: 1,
-            isMobile: false,
-            hasTouch: false,
+            isMobile: true,
+            hasTouch: true,
```
Comment (atabel) [inline src/test-utils/index.tsx:107]: tablets are mobile/touch devices

File: src/__acceptance_tests__/carousel-ssr-acceptance-test.tsx [hunk 4]
```diff
@@ -5,11 +5,12 @@ test('ssr carousel mobile', async () => {
 });
 
 test('ssr carousel tablet', async () => {
-    // TODO WEB-1672: fix hydration mismatches and enable the check
+    // checkHidrationVisualMismatch is disabled because for some reason there are some small visual differences in TABLET
+    // after hidration. Also, in some cases the carousel is scrolled to the next page and this causes a visual mismatch in
+    // the bullets. I've not found a way to know why this happens.
     await openSSRPage({name: 'carousel', device: 'TABLET', checkHidrationVisualMismatch: false});
```
Comment (atabel) [inline src/__acceptance_tests__/carousel-ssr-acceptance-test.tsx:11]: I wasn't able to find a fix for this

**Review by pladaria — APPROVED** [review]

**Review by marcoskolodny — APPROVED** [review]

**Pull request merged**